### ISSUE TITLE
emotion 11 conflicts

### ISSUE THREAD

**Issue by ChrisSargent** [react-theming/storybook-addon#10]
Hi There,

Is this repo and add-on still maintained? I'm having conflicts because this addon has emotion10 as a devDependency but my components are using v11.

Thanks,

Chris

**Comment by UsulPro** [comment]
Hi @ChrisSargent 
How devDependencies can conflict with your project dependencies? It maybe because Storybook itself depends on emotion. How does it work for you in Storybook without this addon?

**Comment by ChrisSargent** [comment]
We have a monorepo, in that monorepo is a component library, that extends / modifies Material UI V5.
Best practice / suggested by Emotion is that this component library only has @emotion as a peer dependency

**Comment by UsulPro** [comment]
Yes this is how it should be ideally. Storybook use emotion to render own components but it shouldn't affect your components

**Comment by ChrisSargent** [comment]
Sorry, I hit submit too soon.

We have a monorepo, in that monorepo is:

- UI Components (a component library, that extends / modifies Material UI V5)
- Application (an application that imports from our UI components library
- Storybook (which also imports from our UI library)


Best practice suggests to keep @emotion as a peer dependency in our component library, such that multiple versions of @emotion are not loaded on the page.

- Your add on (and @storybook/addon-actions requires @storybook/theming) which requires emotion 10, so the @emotion/v10 packages are hoisted to our monorepo root

- Our application has a normal dependency on @emotion/v11, which is also hoisted to the monorepo root

So, to make our application work (i.e. not have 2 versions of emotion loaded), we have to put a resolutions field in our root package to use @emotion v11

But of course, this breaks those storybook add ons :-)

For now, I have stopped using the add-on and now just use a toolbar select, a decorator to switch themes and added the following code to main.js in SB:

```
    config.resolve = {
      ...config.resolve,
      alias: {
        ...config.resolve.alias,
        // @storybook/addon-actions requires @storybook/theming which requires emotion 10
        // But we have to resolve to emotion11 at the root to ensure our components and theme work correctly
        // These lines ensure that webpack knows where to look when storybook is importing the modules it needs
        '@emotion/core': require.resolve('@emotion/react'),
        '@emotion/styled': require.resolve('@emotion/styled'),
        'emotion-theming': require.resolve('@emotion/react'),
      },
    };

```

So, it's not that I'm looking for a solution, more just curious if this packages is updated? Also, I wonder if you would also be able to set @emotion in peer deps, if it would help but also maybe break some things there?

**Comment by UsulPro** [comment]
@ChrisSargent yes this addon is maintained and potentially I can update dependencies. But I can't add emotion as dependencies to this addon because it's not only for emotion. Actually I believe adding emotion as peer dependency doesn't solve what you need. To clarify:
- this addon is addon for theming that's maintained withing React Theming org
- addon-actions is internal Storybook addon and maintained withing Storybook org
- most addons (not only these two) depends on `@storybook/theming` package that's also a part of Storybook inner package and maintains withing Storybook org (and indeed it has Emotion in dependencies)

What I see now is that the issue with Storybook itself rather then with this particular addon. To test you can temporary exclude this addon from Storybook (don't forget to remove it from package.json and reinstall dependencies) if I understand this correctly you still will have issues with addon-actions (or potentially with any other Storybook addon as most of them depends on @storybook/theming). If this true it's worth to raise an issue in Storybook repo as it could be general problem and affect other Emotion users.

Please let me know can I do something on my side with this addon?

I still a bit confused why dependency conflict isn't resolved by your package manager? What Emorion version do you have in your devDependencies?

**Comment by ChrisSargent** [comment]
hmm, I agree I'm not sure why either! Was a late night troubleshooting :-). I think you're probably correct about the @storybook/theming package - and maybe that was causing my issue all along (I think I discovered that one after opening my issue here). And in fact, yes, I still did have this issue after uninstalling this particular addon (well, it was actually the material-ui one you have too but I saw in an issue there that this is now the preferred addon).

FYI, since emotion v11 there is now a console warning in dev mode if more than one instance is found on the page. But basically the error manifests itself as the theme not being injected in to styled components.

I don't have @emotion as a devDependency, this addon has it has a devDependency. The only place I have @emotion as an explicit dependency, is in our application folder (the other is a peerDependency), and is V11.

I'll do some more digging, now I know this is still maintained properly I'd be happy to submit a PR if / when required.
